src/managers/TileManager.ts

```diff
@@ -38,16 +38,72 @@ class TileManager {
                 ),
                 session.seed,
             );
-            this.list.forEach((tile: Tile) => {
-                const state: TileState = TileState.Unused; // TODO
+
+            for (let i = 0; i < this.list.length; i++) {
+                const tile: Tile = this.list[i];
                 const dto: Nullable<TileDTO> =
                     dtos.find((dto: TileDTO) => dto.id === tile.id) || null;
+
+                let state: TileState = this.identifyTiles(
+                    dto,
+                    session.turnPlayer!,
+                    i,
+                );
                 tile.apply(state, dto);
-            });
+            }
+            this.getTilesInHand(); //TODO where to save/use
```

Comment:
You can remove the line 54 since its not used there and also not saved anywhere, its called when needed, e.g. in the hand ui.